### PR TITLE
Feat([RS]):Add management of favorite dashboards

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
**Description**: Add management of favorite dashboards

**Video**

[demoo.webm](https://github.com/user-attachments/assets/dc92c0ea-21e8-41c4-8a7b-68033fb34d36)
 

**Snapshots**: when will merge to the feature branch

### CODE REVIEW

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -17862,3 +17862,18 @@ msgstr "Sticky para cualquier estado que no sea OK"
 
 # msgid "There are errors in the form."
 # msgstr ""
+
+# msgid "Show only dashboards marked as favorites"
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:17866]:
```suggestion
# msgid "Show only dashboards added to favorites"
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -17862,3 +17862,18 @@ msgstr "Sticky para cualquier estado que no sea OK"
...
+# msgstr ""
+
+# msgid "The dashboard is successfully marked as favorite"
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:17869]:
```suggestion
# msgid "The dashboard has been added to favorites."
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -17862,3 +17862,18 @@ msgstr "Sticky para cualquier estado que no sea OK"
...
+# msgstr ""
+
+# msgid "The dashboard is successfully marked as unFavorite"
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:17872]:
```suggestion
# msgid "The dashboard has been removed from favorites."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -19617,3 +19617,19 @@ msgstr "minute"
 
 msgid "There are errors in the form."
 msgstr "Il y a des erreurs dans le formulaire."
+
+msgid "Show only dashboards marked as favorites"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19621]:
```suggestion
msgid "Show only dashboards added to favorites"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -19617,3 +19617,19 @@ msgstr "minute"
...
+msgstr "Afficher uniquement les tableaux de bord marqués comme favoris"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19622]:
```suggestion
msgstr "Afficher uniquement les tableaux de bord ajoutés aux favoris"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -19617,3 +19617,19 @@ msgstr "minute"
...
+
+msgid "The dashboard is successfully marked as favorite"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19624]:
```suggestion
msgid "The dashboard has been added to favorites."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -19617,3 +19617,19 @@ msgstr "minute"
...
+msgstr "Le tableau de bord a été marqué comme favori avec succès"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19625]:
```suggestion
msgstr "Le tableau de bord a été ajouté aux favoris."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -19617,3 +19617,19 @@ msgstr "minute"
...
+
+msgid "The dashboard is successfully marked as unFavorite"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19627]:
```suggestion
msgid "The dashboard has been removed from favorites."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -19617,3 +19617,19 @@ msgstr "minute"
...
+msgstr "Le tableau de bord a été marqué comme non favori avec succès"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19628]:
```suggestion
msgstr "Le tableau de bord a été retiré des favoris."
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -18323,3 +18323,18 @@ msgstr ""
 
 # msgid "There are errors in the form."
 # msgstr ""
+
+# msgid "Show only dashboards marked as favorites"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:18327]:
```suggestion
# msgid "Show only dashboards added to favorites"
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -18323,3 +18323,18 @@ msgstr ""
...
+# msgstr ""
+
+# msgid "The dashboard is successfully marked as favorite"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:18330]:
```suggestion
# msgid "The dashboard has been added to favorites."
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 12]
```diff
@@ -18323,3 +18323,18 @@ msgstr ""
...
+# msgstr ""
+
+# msgid "The dashboard is successfully marked as unFavorite"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:18333]:
```suggestion
# msgid "The dashboard has been removed from favorites."
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -18303,3 +18303,18 @@ msgstr "Você não salvou as alterações, deseja continuar?"
 
 # msgid "There are errors in the form."
 # msgstr ""
+
+# msgid "Show only dashboards marked as favorites"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:18307]:
```suggestion
# msgid "Show only dashboards marked as favorites"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 14]
```diff
@@ -18303,3 +18303,18 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+# msgstr ""
+
+# msgid "The dashboard is successfully marked as favorite"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:18310]:
```suggestion
# msgid "The dashboard has been added to favorites."
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -18303,3 +18303,18 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+# msgstr ""
+
+# msgid "The dashboard is successfully marked as unFavorite"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:18313]:
```suggestion
# msgid "The dashboard has been removed from favorites."
```

File: centreon/www/front_src/src/Dashboards/translatedLabels.tsx [hunk 16]
```diff
@@ -72,3 +72,10 @@ export const labelSaveYourDashboardForThumbnail =
   'Save your dashboard to generate a thumbnail';
 export const labelDataDisplayedForRepresentativeUse =
   'Thumbnails show a snapshot of your data, taken when the dashboard is saved';
+export const labelFavoriteFilter = 'Show only dashboards marked as favorites';
```
Comment (smau-centreon) [inline centreon/www/front_src/src/Dashboards/translatedLabels.tsx:75]:
```suggestion
export const labelFavoriteFilter = 'Show only dashboards added to favorites';
```

File: centreon/www/front_src/src/Dashboards/translatedLabels.tsx [hunk 17]
```diff
@@ -72,3 +72,10 @@ export const labelSaveYourDashboardForThumbnail =
...
+export const labelDashboardSuccessfullyMarkedAsFavorite =
+  'The dashboard is successfully marked as favorite';
```
Comment (smau-centreon) [inline centreon/www/front_src/src/Dashboards/translatedLabels.tsx:77]:
```suggestion
  'The dashboard has been added to favorites.';
```

File: centreon/www/front_src/src/Dashboards/translatedLabels.tsx [hunk 18]
```diff
@@ -72,3 +72,10 @@ export const labelSaveYourDashboardForThumbnail =
...
+export const labelDashboardSuccessfullyMarkedAsUnFavorite =
+  'The dashboard is successfully marked as unFavorite';
```
Comment (smau-centreon) [inline centreon/www/front_src/src/Dashboards/translatedLabels.tsx:79]:
```suggestion
  'The dashboard has been removed from favorites.';
```

File: centreon/www/front_src/src/Dashboards/translatedLabels.tsx [hunk 19]
```diff
@@ -72,3 +72,10 @@ export const labelSaveYourDashboardForThumbnail =
   'Save your dashboard to generate a thumbnail';
 export const labelDataDisplayedForRepresentativeUse =
   'Thumbnails show a snapshot of your data, taken when the dashboard is saved';
+export const labelFavoriteFilter = 'Show only dashboards added to favorites';
+export const labelDashboardAddedToFavorites =
+  'The dashboard has been added to favorites.';
```
Comment (cg-tw) [inline centreon/www/front_src/src/Dashboards/translatedLabels.tsx:77]:
```suggestion
  'The dashboard has been added to your favorites.';
```

File: centreon/www/front_src/src/Dashboards/translatedLabels.tsx [hunk 20]
```diff
@@ -72,3 +72,10 @@ export const labelSaveYourDashboardForThumbnail =
...
+export const labelDashboardRemovedFromFavorites =
+  'The dashboard has been removed from favorites.';
```
Comment (cg-tw) [inline centreon/www/front_src/src/Dashboards/translatedLabels.tsx:79]:
```suggestion
  'The dashboard has been removed from your favorites.';
```

**Review by cg-tw — APPROVED** [review]

**Review by quborg — APPROVED** [review]

File: centreon/www/front_src/src/Dashboards/api/models.ts [hunk 21]
```diff
@@ -69,6 +69,7 @@ export type Dashboard = NamedEntity & {
     type: 'global' | 'manual';
     interval: number | null;
   };
+  isFavorite?: boolean; // a enlever optional
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/api/models.ts:72]: What is the point on this?

Comment (Noha-ElAbrouki) [reply centreon/www/front_src/src/Dashboards/api/models.ts]: was a reminder to delete the ?

**Review by Thebarda — APPROVED** [review]

**Pull request merged**